frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/utils.ts

```diff
@@ -174,14 +178,17 @@ export function useTableLoader(path: TreePath) {
                 })),
         })),
       );
-      setTree((current) => merge(current, newTree));
+      setTree((current) => {
+        const merged = merge(current, newTree);
+        return _.isEqual(current, merged) ? current : merged;
+      });
     },
     [getDatabases, getSchemas, getTables],
   );
 
   useDeepCompareEffect(() => {
     load(path);
-  }, [load, path, tables.isFetching]);
+  }, [load, path]);
```

Comment:
This extra dep was causing an infinite data-loading loop (there's an assertion for it in one of the tests)